source/docs/software/basic-programming/git-getting-started.rst

```diff
@@ -0,0 +1,250 @@
+
+Run: ``git checkout branch-name`` where branch-name is the branch that was previously created.
+
+Merging
+-------
+
+In scenarios where you want to copy one branches history into another, you can merge them. A merge is done by calling ``git merge branch-name`` with branch-name being the name of the branch to merge from. It is automatically merged in the current active branch. 
+
+It's common for a remote repository to contain work (history) that you do not have. Whenever you run ``git pull``, it will attempt to automatically merge those commits. That merge may look like the below.
+
+.. graphviz::
+
+   digraph branches {
+      "Example Repo" [ shape=cylinder]
+      FeatureA [ shape=ellipse]
+      FeatureB [ shape=ellipse]
+      FeatureC [ shape=ellipse]
+      "Example Repo" -> FeatureA
+      "Example Repo" -> FeatureB
+      "Example Repo" -> FeatureC
+      "Update File 1" [ shape=box]
+      FeatureA -> "Update File 1"
+      "Update File 2" [ shape=box]
+      FeatureB -> "Update File 2"
+      "Update File 3" [ shape=box]
+      FeatureC -> "Update File 3"
+   }
+
+However, in the above example, what if File 1 was modified by both branch FeatureA and FeatureB? This is called a **merge conflict**. A merge conflict will can be resolved by editing the conflicting file. In the example, we would need to edit File 1 to keep the history or changes that we want. After that has been done. Simply re-add, re-commit, and then push your changes.
+
+Resets
+------
+
+Sometimes history needs to be reset, or a commit needs to be undone. This can be done multiple ways. 
+
+Reverting the Commit
+^^^^^^^^^^^^^^^^^^^^
+
+.. note:: You cannot revert a merge, as git does not know which branch or origin it should choose.
+
+To revert history leading up to a commit run ``git revert commit-id``. The commit IDs can be shown using the ``git log`` command. 
+
+Resetting the Head
+^^^^^^^^^^^^^^^^^^
+
+.. warning:: Forcibly resetting the head is a dangerous command. It permanently erases all history past the target. You have been warned!
+
+Run: ``git reset --hard commit-id``. 
+
+Forks
+-----
+
+Forks can be treated similarly to branches. You can merge the upstream (original repository) into the origin (forked repository). 
+
+Updating a Fork
+^^^^^^^^^^^^^^^
+
+1. Add the upstream: ``git remote add upstream https://github.com/ORIGINAL_OWNER/ORIGINAL_REPOSITORY.git``
+2. Confirm it was added via: ``git remote -v``
+3. Pull changes from upstream: ``git fetch upstream``
+4. Merge the changes into head: ``git merge upstream/upstream-branch-name``
+
+Gitignore
+---------
+
+.. important:: It is extremely important that teams **do not** modify the .gitignore file that is included with their robot project. This can lead to offline deployment not working.
+
+A .gitignore file is commonly used as a list of files to not automatically commit with ``git add .``. Any files or directory listed in this file will **not** be commited. 
+
+Hiding a Folder
```

Comment:
I recommend linking https://www.atlassian.com/git/tutorials/saving-changes/gitignore

It explains gitignore pattern matching well